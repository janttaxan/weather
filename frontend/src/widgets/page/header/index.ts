export { Header } from 'widgets/page/header/header';
