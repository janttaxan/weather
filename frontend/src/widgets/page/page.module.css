@import 'shared/variables.css';

.root {
  width: 100%;
  height: 100%;

  display: grid;
  grid-row-gap: 32px;
  grid-column-gap: 0;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr auto;
  grid-template-areas:
    'h'
    'm'
    'f';
}

.header {
  grid-area: h;
}

.main {
  grid-area: m;
  padding: 0 16px;
  width: 100%;
  max-width: $maxWidthContent;
  margin: 0 auto;
}

.footer {
  grid-area: f;
  padding: 16px;

  display: flex;
  align-items: center;
  justify-content: center;

  background-color: rgb($colorLightBlack);
}

@media (min-width: $breakpointSmall) {
  .root {
    grid-row-gap: 40px;
    grid-column-gap: 16px;
    grid-template-columns: repeat(12, 1fr);
    grid-template-areas:
    'h h h h h h h h h h h h'
    'm m m m m m m m m m m m'
    'f f f f f f f f f f f f';
  }

  .main {
    padding: 0 40px;
  }

  .footer {
    padding: 24px 40px;
  }
}
