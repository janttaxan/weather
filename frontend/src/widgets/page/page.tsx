import styles from './page.module.css';

import { ReactElement } from 'react';

import { CommonContext } from 'app-layers/common-context/types';

import { Header } from 'widgets/page/header';

interface PageProps {
  context: CommonContext;
  children: ReactElement;
}

export const Page = (props: PageProps) => {
  return (
    <div className={styles.root}>
      <Header />
      <main className={styles.main}>{props.children}</main>
      <footer className={styles.footer}>footer</footer>
    </div>
  );
};
