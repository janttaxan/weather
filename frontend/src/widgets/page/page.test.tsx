import { render, screen } from '@testing-library/react';
import { Page } from 'widgets/page/page';

describe('Page', () => {
  test('render children', () => {
    const { container } = render(
      <Page context={{ user: null }}>
        <div data-testid='child'>
          <h1>Heading</h1>
        </div>
      </Page>
    );
    expect(screen.getByTestId('child')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
