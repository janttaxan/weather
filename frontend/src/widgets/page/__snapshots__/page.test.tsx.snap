// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page render children 1`] = `
<div>
  <div
    class="root"
  >
    <header
      class="root"
    >
      <div
        class="root"
      >
        <input
          class="input"
          placeholder="Поиск"
          type="text"
        />
      </div>
      <div>
        navigation
      </div>
    </header>
    <main
      class="main"
    >
      <div
        data-testid="child"
      >
        <h1>
          Heading
        </h1>
      </div>
    </main>
    <footer
      class="footer"
    >
      footer
    </footer>
  </div>
</div>
`;
